src/index.ts

```diff
@@ -0,0 +1,115 @@
+interface IToDoList {
+  get tasks(): Task[];
+  getTask(id: TaskId): Task | null;
+  addTask(name: string, body: string, isRequireEditConfirm?: boolean): void;
+  removeTask(id: TaskId): void;
+  editTask(id: TaskId, data: { name: string; body: string }): void;
+  toogleCompleted(id: TaskId): void;
+  getTasksAmount(): number;
+  getNotCompletedAmount(): number;
+  searchTasks(value: string): Task[];
```

Comment:
There is the requirement: "Separately, it is necessary to expand the behavior of the list and add the possibility of searching for a note by name or content."

It means that it should be a separate class for that functionality.